visualize-admin/visualization-tool
# feat: Map chart improvements

Closes #1425

This PR:
- consolidates some of the logic related to initializing map color fields,
- adds opacity property to every color field (and enables it via slider for all symbol color types),
- fixes empty map charts when symbol layer is the only enabled layer and is based on geometries, and not coordinates,
- removes unused header progress code,
- increases radius of circle legend elements,
- adds black border to map symbols,
- fixes `useMemo` issues with chart data and coordinates computations.

## How to test
### PR
1. Go to [this link](https://visualization-tool-git-feat-improve-map-symbols-and-legend-ixt1.vercel.app/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/9&dataSource=Prod).
2. Switch to a map chart.
3. Remove Areas (change dimension to `None`).
4. Add Symbols (change dimension to `Kanton`).
5. ✅ See that the map is shown.
6. ✅ See that each symbol is encircled with a black border.
7. Change color measure to `Leistung in kW`.
8. ✅ See that it's possible to control the opacity with a slider.
9. Change color measure to `Kanton`.
10. ✅ See that it's possible to control the opacity with a slider.
11. ✅ See that circles in the legend have a size that matches the height of the text.

### TEST
1. Go to [this link](https://test.visualize.admin.ch/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/9&dataSource=Prod).
2. Switch to a map chart.
3. Remove Areas (change dimension to `None`).
4. Add Symbols (change dimension to `Kanton`).
5. ❌ See that the map is not shown.
6. Add the Areas back.
7. Switch to Symbols options.
8. ❌ See that symbols are not encircled with black borders.
9. Change color measure to `Leistung in kW`.
10. ❌ See that it's not possible to control the opacity.
11. Change color measure to `Kanton`.
12. ❌ See that it's not possible to control the opacity.
13. ❌ See that circles in the legend have a size that is smaller than the height of the text.
